entity/contents/snapshot.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include "flutter/fml/macros.h"
+#include "impeller/geometry/matrix.h"
+#include "impeller/geometry/rect.h"
+#include "impeller/renderer/texture.h"
+
+namespace impeller {
+
+class ContentContext;
+class Entity;
+
+/// Represents a screen space texture and its intended draw position.
+struct Snapshot {
+  std::shared_ptr<Texture> texture;
+  /// The offset from the origin where this texture is intended to be
+  /// rendered.
+  Vector2 position;
+
+  static std::optional<Snapshot> FromTexture(const ContentContext& renderer,
```

Comment:
It is hard to tell from just the name `FromTexture` and its arguments about what this method is meant to do. Can the method name be more descriptive and also please document it.

Reply:
Changed name to FromTransformedTexture/added a comment.

This will no longer be used/can probably be removed once I do the smarter UV mapping changes I mentioned in other comments BTW.